### ISSUE TITLE
Vertices in fsaverage4 are shuffled compared to fsaverage_k>4

### ISSUE THREAD

**Issue by sdascoli** [nilearn/nilearn#5693]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Operating system

- [ ] Linux
- [ ] Mac
- [ ] Windows

### Python version

- [ ] 3.13
- [ ] 3.12
- [ ] 3.11
- [ ] 3.10

### nilearn version

0.12.1

### Expected behavior

```python
from nilearn.datasets import load_fsaverage
import torch
import matplotlib.pyplot as plt

fig, axarr = plt.subplots(1,4, figsize=(10,3))

for k in [3,4,5,6]:
    small = load_fsaverage(f"fsaverage{k}")
    large = load_fsaverage(f"fsaverage{k+1}")

    # choose a surface (white, pial, sphere… same topology, same vertex order)
    x = small.sphere.parts["left"].coordinates
    y = large.sphere.parts["left"].coordinates

    x = torch.from_numpy(x)
    y = torch.from_numpy(y)

    distances = torch.cdist(x, y)
    print(distances.shape)
    indices = torch.argmin(distances, dim=1)
    axarr[k-3].plot(indices)
    axarr[k-3].set_title(f"k={k}")
    axarr[k-3].set_xlabel("Vertex index in fsaverage{k}")
    axarr[k-3].set_ylabel("Vertex index in fsaverage{k+1}")
plt.tight_layout()
plt.show()
```

Since we expect that fsaverage_k should be a prefix of fsaverage_k+1, we expect that when taking
fsaverage_k[:size(fsaverage_k-1)], we should recover the vertices of fsaverage_k-1.

### Current behavior & error messages

This indeed works for k=3,5,6, but not for k=4 as shown in the figure!
This means that downsampling from one mesh to another can lead to vertices being mixed up.
Is this a bug?

<img width="831" height="244" alt="Image" src="https://github.com/user-attachments/assets/3832d8c6-345d-4e18-9b44-5c15827a1bd7" />

### Steps and code to reproduce bug

```python
# Paste your code here


```

**Comment by bthirion** [comment]
Thx for reporting.
I guess that the issue comes from the public version of fsaverage, but this should be checked.

**Comment by Remi-Gau** [comment]
- Isn't this the same problem as the one in: https://github.com/nilearn/nilearn/issues/3415?

**Comment by sdascoli** [comment]
Indeed, I missed that one thanks!
